### PR TITLE
Preventing empty extensions to be sent into telemetry

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Preventing empty extensions to be sent into telemetry, for the cases such as the following (three empty extension metrics) - 

```
{
  "PerfMetrics": {
    "Microsoft.OSTCExtensions.OSPatchingForLinux-2.0.0.5": {},
    "Microsoft.Azure.Extensions.CustomScript-2.0.6": {},
    "Microsoft.Azure.Diagnostics.LinuxDiagnostic-3.0.121": {},
    "Microsoft.OSTCExtensions.LinuxDiagnostic-2.3.9029": {
      "cpu": {
        "cur_cpu": [...]
      },
      "memory": {
        "cur_mem": [...],
        "max_mem": [...]
      }
    },
    "walinuxagent.service": {
      "cpu": {
        "cur_cpu": [...]
      },
      "memory": {
        "cur_mem": [...],
        "max_mem": [...]
      }
    },
    "Microsoft.OSTCExtensions.VMAccessForLinux-1.4.7.1": {
      "cpu": {
        "cur_cpu": [...]
      },
      "memory": {
        "cur_mem": [...],
        "max_mem": [...]
      }
    }
  },
  "SchemaVersion": 1.0
}
```

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1557)

### CODE REVIEW

File: tests/common/test_cgroupstelemetry.py [hunk 1]
```diff
@@ -313,7 +313,8 @@ def test_telemetry_calculations(self, *args):
             self.assertListEqual(generate_metric_list(cpu_percent_values),
                                  collected_metrics[name]["cpu"]["cur_cpu"][0:5])
 
-    def test_cgroup_tracking(self):
+    @patch("azurelinuxagent.common.osutil.default.DefaultOSUtil._get_proc_stat")
```
Comment (narrieta) [inline tests/common/test_cgroupstelemetry.py]: why do we need to mock getprocstat? thanks

Comment (vrdmr) [reply tests/common/test_cgroupstelemetry.py]: This particular unit test does not need the actual data from `/proc/stat` and to make it run on Mac, I mocked the call.

Comment (narrieta) [reply tests/common/test_cgroupstelemetry.py]: would you add a comment explaining that? thanks

Comment (vrdmr) [reply tests/common/test_cgroupstelemetry.py]: Done.

File: tests/common/test_cgroupstelemetry.py [hunk 2]
```diff
@@ -313,7 +313,10 @@ def test_telemetry_calculations(self, *args):
             self.assertListEqual(generate_metric_list(cpu_percent_values),
                                  collected_metrics[name]["cpu"]["cur_cpu"][0:5])
 
-    def test_cgroup_tracking(self):
+    # mocking get_proc_stat to make it run on Mac and other systems
```
Comment (narrieta) [inline tests/common/test_cgroupstelemetry.py]: it'd be nice to add the comment to all instances of the mock

**Pull request merged**